bool reset_path
---
bool success
